src/test/ui/feature-gates/underscore_const_names_feature_gate.rs

```diff
@@ -9,6 +9,5 @@
 // except according to those terms.
 
 const _: () = (); //~ ERROR is unstable
-static _: () = (); //~ ERROR is unstable
```

Comment:
I understand this wasn't in the RFC, but I see no reason not to support it.

Reply:
`const _: T = expr;` was permitted because it has a reasonable generalization to irrefutable patterns (e.g. `const (A, B) = (1, 2);`) but does `static` also have that with the same-address notion?

Reply:
At least the motivation in https://github.com/rust-lang/rfcs/pull/2526 does not apply to statics.
Constants don't even exist in MIR (AFAIK), so they cannot be used for compile-time asserts, while statics end up in the resulting executable/library, or need to be optimized away.